src/declarativetabmodel.cpp

```diff
@@ -249,7 +250,8 @@ const QList<Tab> &DeclarativeTabModel::tabs() const
 
 const Tab &DeclarativeTabModel::activeTab() const
 {
-    return m_activeTab;
+    Q_ASSERT(contains(m_activeTabId));
```

Comment:
Probably with a connection to the active model we could remove ```m_tabId``` from DeclarativeWebContainer as well and just leave tabId() method to return activate tab id from the current active model. It is used in ```DeclarativeWebContainer::reload``` and in WebView.qml within the WebPage {} to check if this is the active tab.

Maybe at some point we could add an attached property like ```isCurrentItem``` to the webpage.